src/hiveboard_bridge/src/TCPServerMonitor.cpp

```diff
@@ -0,0 +1,41 @@
+#include "../include/hiveboard_bridge/TCPServerMonitor.h"
+
+TCPServerMonitor::TCPServerMonitor() {
+    m_maxFd = 0;
+    m_serverFd = 0;
+    FD_ZERO(&m_fileDescriptorSet);
+}
+
+TCPServerMonitor::~TCPServerMonitor() {}
+
+void TCPServerMonitor::addServer(int serverFd) {
+    m_serverFd = serverFd;
+    addConn(m_serverFd);
+}
+
+void TCPServerMonitor::addConn(int conn) {
+    FD_SET(conn, &m_fileDescriptorSet);
+
+    if (conn > m_maxFd) {
+        m_maxFd = conn;
+    }
+}
+
+void TCPServerMonitor::removeConn(int conn) {
+    FD_CLR(conn, &m_fileDescriptorSet);
+    m_maxFd = m_serverFd; // decrement the max since we closed a socket.
```

Comment:
Comment still linked to code? How does this decrement the max?

Reply:
I edited the comment so that it is less confusing. The idea here is that `select` needs the user to keep track of the maximal file descriptor (FD) value that is monitored (this is quite an old lib so the design choices seem dated a bit).

Anyway, when we create a client socket, there is a new FD that gets created. That new value must be assigned to the `m_maxFd`, thus overwriting the previous value (set initially to the server FD). 

If I close the socket, I need to remove the value from `m_maxFd` and restore its value to the server FD. Since this code can only manage a single client, "decrementing" means restoring the value to the server FD.